From 4887e460d59fd1e3714409a0f40a9a1bbf5e838b Mon Sep 17 00:00:00 2001
From: Bassel Saba <basselsa@marvell.com>
Date: Sun, 11 Oct 2015 17:16:07 +0300
Subject: [PATCH 0013/1345] fix: sdhci-pxav3: Disable clock inversion for HS
 SD cards

commit  7cdb5206fda744fd5bfd6089f26b50dc3fc04ca0 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The Clock inversion option is needed to support slow frequencies
when the card input hold time requirement is high.
This setting is not required for high speed SD and might cause timing
violation.
Erratum 'FE-2946959' was recently updated to specify this HS SD setting

Change-Id: I4d91f0707e79cf1fb4a392ea9424cbf647a8f6ed
Signed-off-by: Bassel Saba <basselsa@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23934
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27302
Tested-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mmc/host/sdhci-pxav3.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-pxav3.c b/drivers/mmc/host/sdhci-pxav3.c
index d943504..b70d725 100644
--- a/drivers/mmc/host/sdhci-pxav3.c
+++ b/drivers/mmc/host/sdhci-pxav3.c
@@ -347,7 +347,8 @@ static void pxav3_set_uhs_signaling(struct sdhci_host *host, unsigned int uhs)
 		    uhs == MMC_TIMING_UHS_DDR50) {
 			reg_val &= ~SDIO3_CONF_CLK_INV;
 			reg_val |= SDIO3_CONF_SD_FB_CLK;
-		} else if (uhs == MMC_TIMING_MMC_HS) {
+		} else if (uhs == MMC_TIMING_MMC_HS ||
+			   uhs == MMC_TIMING_SD_HS) {
 			reg_val &= ~SDIO3_CONF_CLK_INV;
 			reg_val &= ~SDIO3_CONF_SD_FB_CLK;
 		} else {
-- 
1.7.9.5

